src/android/src/com/chariotsolutions/nfc/plugin/NfcPlugin.java

```diff
@@ -650,9 +682,169 @@ private void fireNdefEvent(String type, Ndef ndef, Parcelable[] messages) {
+    {
+        CordovaWebView webview = this.webView;
+        new AsyncTask<Tag,Void,String>()
+        {
+            protected String doInBackground(Tag... tags)
+            {
+                int sector = -1;
+                Tag tag = tags[0];
+                JSONObject tagJSON = Util.tagToJSON(tag);
+                ByteArrayOutputStream out = new ByteArrayOutputStream();
+
+                MifareClassic mifare = MifareClassic.get(tag);
+                int size = mifare.getSize();
+                int type = mifare.getType();
+                try
+                {
+                    tagJSON.put("type",type);
+                    tagJSON.put("size",size);
+                }
+                catch (JSONException e)
+                {
+                    e.printStackTrace();
+                }
+
+                try
+                {
+                    mifare.connect();
+                    mifare.setTimeout(3000);
+
+                    for(int i = 0; i < mifare.getBlockCount(); ++i)
+                    {
+                        if(mifare.blockToSector(i) != sector)
+                        {
+                            sector = mifare.blockToSector(i);
+                            mifare.authenticateSectorWithKeyA(sector,mifare.KEY_DEFAULT);
+                            //mifare.authenticateSectorWithKeyB(sector,mifare.KEY_DEFAULT);
+                        }
+                        byte[] block = mifare.readBlock(i);
+                        out.write(block);
+                    }
+                }
+                catch (IOException e)
+                {
+                    e.printStackTrace();
+                    try
+                    {
+                        tagJSON.put("error",e.getMessage());
+                    }
+                    catch (JSONException e1)
+                    {
+                        e1.printStackTrace();
+                    }
+
+                    try
+                    {
+                        mifare.close();
+                    }
+                    catch (IOException e1)
+                    {
+                        e1.printStackTrace();
+                    }
+                }
+
+            try
+            {
+                if(out.size() > 0)
+                {
+                    tagJSON.put("data", Util.byteArrayToJSON(out.toByteArray()));
+                }
+            }
+            catch (JSONException e)
+            {
+                e.printStackTrace();
+            }
+
+            String command = MessageFormat.format(javaScriptEventTemplate, MIFARE_CLASSIC, tagJSON);
+            Log.v(TAG, command);
+            return command;
+            }
+
+            @Override
+            protected void onPostExecute(String command)
+            {
+                webView.sendJavascript(command);
+            }
+        }.execute(tag);
+    }
+
+    private void fireNdefFormatableEvent (Tag tag) {
+        JSONObject tagJSON = Util.tagToJSON(tag);
+        /*
```

Comment:
what's the reason for this being commented out? either keep, remove or add a comment on why it is commented out